prometheus/gc_collector.go

```diff
@@ -0,0 +1,43 @@
+package prometheus
+
+import (
+	"runtime/debug"
+)
+
+type gcCollector struct {
+	numGC      Gauge
+	pauseTotal Gauge
+}
+
+// NewGCCollector returns a collector which exports metrics about the
+// garbage collection stats of the current process.
+func NewGCCollector() *gcCollector {
+	return &gcCollector{
+		numGC: NewGauge(GaugeOpts{
+			Name: "process_gc_num_total",
+			Help: "Total number of garbage collector incovations for the current process.",
```

Comment:
The _count suffix is a bit tricky at it uses the C word... I prefer to have counters always end on _total, but for the counters that are part of a summary  that would be confusing. So I tolerate _count and _sum in a summary.

In this case, I think putting both in a summary without quantiles will be formally more correct, but won't help understand the metrics ("Count of pauses?" - when in is just the count of GCs).

In any case, those values should be counters, not gauges.